### PR TITLE
[FIXED JENKINS-39837] Set the GithubWeb browser when creating SCM

### PR DESCRIPTION
**Pull request by stephenc**
See [JENKINS-39837](https://issues.jenkins-ci.org/browse/JENKINS-39837)

@reviewbybees

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java [hunk 1]
```diff
@@ -976,6 +1028,8 @@ protected boolean isCategoryEnabled(@NonNull SCMHeadCategory category) {
         StandardCredentials credentials = Connector.lookupScanCredentials(getOwner(), apiUri, scanCredentialsId);
         GitHub hub = Connector.connect(apiUri, credentials);
         GHRepository repo = hub.getRepository(getRepoOwner() + '/' + repository);
+        repositoryUrl = repo.getHtmlUrl();
```
Comment (rsandell) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java]: These looks like a strange side effect of asking for actions.

**Comment by rsandell** [comment]
:bee:

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java (hunk 1 above)
Comment (stephenc) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java]: refresh the cache every time you can

Comment (stephenc) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java]: However the next line was a bug as the next line triggers a remote request (oh thank you GitHub API by KK for not clearly delineating the methods that cause a remote request from the methods that do not

**Pull request merged**